### PR TITLE
fix(sdk-core): support backup party id in mpcv2 signing utils

### PR DESCRIPTION
**Pull request by islamaminBitGo**
TICKET: HSM-442

### CODE REVIEW

**Review by mohammadalfaiyazbitgo — COMMENTED** [review]
any unit tests you can add for this?

File: modules/sdk-core/src/bitgo/tss/ecdsa/ecdsaMPCv2.ts [hunk 1]
```diff
@@ -17,7 +17,8 @@ import assert from 'assert';
 
 export async function getSignatureShareRoundOne(
   round1Message: DklsTypes.DeserializedBroadcastMessage,
-  userGpgKey: openpgp.SerializedKeyPair<string>
+  userGpgKey: openpgp.SerializedKeyPair<string>,
+  partyId: 0 | 1 = 0
```
Comment (mohammadalfaiyazbitgo) [inline modules/sdk-core/src/bitgo/tss/ecdsa/ecdsaMPCv2.ts:21]: any reason you're using `0 | 1` instead of `boolean` ?

**Comment by islamaminBitGo** [comment]
> any unit tests you can add for this?

there is no value of any additional tests in my opinion, the user signing is tested so adding a backup signing test would be a clone of that

**Review by pranavjain97 — DISMISSED** [review]
Changes look good. Can you add a test with the backup id case here `tssUtils/ecdsaMPCv2/signTxRequest.ts`?

**Comment by pranavjain97** [comment]
> > any unit tests you can add for this?
> 
> there is no value of any additional tests in my opinion, the user signing is tested so adding a backup signing test would be a clone of that

It's more so that nobody removes that logic from this code if there is test for it

**Comment by alebusse** [comment]
do we want to change the top method signRequestBase too https://github.com/BitGo/BitGoJS/blob/a481acddd572873aa9d3303f97832dbe88289de8/modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsaMPCv2.ts#L698?

**Comment by islamaminBitGo** [comment]
> do we want to change the top method signRequestBase too
> 
> https://github.com/BitGo/BitGoJS/blob/a481acddd572873aa9d3303f97832dbe88289de8/modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsaMPCv2.ts#L698
> 
> ?

i feel this might be out of scope for this PR since this is meant only for the utils that other apps are using rather than the hot wallet regular flow itself. Nevertheless, yes, I think we can do this but in a follow up ticket + PR

**Comment by islamaminBitGo** [comment]
actuall

> > do we want to change the top method signRequestBase too
> > https://github.com/BitGo/BitGoJS/blob/a481acddd572873aa9d3303f97832dbe88289de8/modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsaMPCv2.ts#L698
> > 
> > ?
> 
> i feel this might be out of scope for this PR since this is meant only for the utils that other apps could be using (such as WRW) rather than the hot wallet regular flow itself. Nevertheless, yes, I think we can do this but in a follow up ticket + PR

actually on second thought, it's not a big change so i think it's ok to include it in this ticket. will apply this change as well

**Comment by islamaminBitGo** [comment]
> Changes look good. Can you add a test with the backup id case here `tssUtils/ecdsaMPCv2/signTxRequest.ts`?

done

**Comment by islamaminBitGo** [comment]
> do we want to change the top method signRequestBase too
> 
> https://github.com/BitGo/BitGoJS/blob/a481acddd572873aa9d3303f97832dbe88289de8/modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsaMPCv2.ts#L698
> 
> ?

done

**Review by mohammadalfaiyazbitgo — APPROVED** [review]

**Pull request merged**